### PR TITLE
Add replace command

### PR DESCRIPTION
**Pull request by rlmcpherson**
Adds a replace command `kops replace` that takes a file argument and replaces existing resources. Supports cluster, instance-group, and federation. This commands allows non-interactive changes to existing clusters. The implementation is similar to the `kops create` command.

The flow is the same as https://github.com/kubernetes/kops/blob/master/docs/changing_configuration.md except using `replace` in lieu of `edit`.

If there's missing documentation that needs to be added, I'm happy to create it.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1066)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Is there a reason that we would not just add the functionality to edit? What does kubectl do? edit -f / update -f, love the idea a ton, and thanks for the work. Will review the code as well.

File: cmd/kops/replace.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename to use to replace the resource")
+	cmd.MarkFlagRequired("filename")
+
+	return cmd
+}
+
+func RunReplace(f *util.Factory, cmd *cobra.Command, out io.Writer, c *UpdateOptions) error {
+	clientset, err := f.Clientset()
+	if err != nil {
+		return err
```
Comment (chrislovecnm) [inline cmd/kops/replace.go]: Can we fmt the err?

File: cmd/kops/replace.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+		default:
+			glog.V(2).Infof("Type of object was %T", v)
+			return fmt.Errorf("Unhandled kind %q in %q", gvk, f)
+		}
+
+	}
+
+	return nil
+}
```
Comment (chrislovecnm) [inline cmd/kops/replace.go]: We able to do some unit tests? We are striving to improve code coverage ;)

**Comment by justinsb** [comment]
Going to merge; this looks good @rlmcpherson .  Also will help with our testing!

**Comment by rlmcpherson** [comment]
Thanks. I'm happy to help with refactoring this. 

As far as the current behavior, I was trying to mirror the `kubectl replace` functionality, which does not have subcommands, just takes a filename flag. 

As far as `edit` vs. `replace`, the `edit` command, at least on kubectl, always opens an editor even when a file is specified so it won't work for non-interactive changes, which was one of the goals with adding this.

**Pull request merged**